internal/abi/fastly/kvstore_guest.go

```diff
@@ -10,166 +10,343 @@ import (
+//              (param $key string)
+//              (param $lookup_config_mask $kv_lookup_config_options)
+//              (param $lookup_configuration (@witx pointer $kv_lookup_config))
+//              (param $handle_out (@witx pointer $kv_store_lookup_handle))
+//              (result $err (expected (error $fastly_status)))
+//          )
+//
+
+//go:wasmimport fastly_kv_store lookup
 //go:noescape
-func fastlyObjectStoreLookup(
-	h objectStoreHandle,
+func fastlyKVStoreLookup(
+	h kvstoreHandle,
 	keyData prim.Pointer[prim.U8], keyLen prim.Usize,
+	mask kvLookupConfigMask,
+	config prim.Pointer[kvLookupConfig],
+	lookupHandle prim.Pointer[kvstoreLookupHandle],
+) FastlyStatus
+
+// Lookup returns a handle to a pending lookup operation
+func (kv *KVStore) Lookup(key string) (kvstoreLookupHandle, error) {
+	keyBuffer := prim.NewReadBufferFromString(key).Wstring()
+
+	// empty
+	var opts kvLookupConfigMask
+	var conf kvLookupConfig
+
+	var lookupHandle kvstoreLookupHandle = invalidKVLookupHandle
+
+	if err := fastlyKVStoreLookup(
+		kv.h,
+		keyBuffer.Data, keyBuffer.Len,
+		opts,
+		prim.ToPointer(&conf),
+		prim.ToPointer(&lookupHandle),
+	).toError(); err != nil {
+		return 0, err
+	}
+
+	if lookupHandle == invalidKVLookupHandle {
+		return 0, FastlyStatusError.toError()
+	}
+
+	return lookupHandle, nil
+}
+
+// witx:
+//
+//          (@interface func (export "lookup_wait")
+//              (param $handle $kv_store_lookup_handle)
+//              (param $body_handle_out (@witx pointer $body_handle))
+//              (param $metadata_buf (@witx pointer (@witx char8)))
+//              (param $metadata_buf_len (@witx usize))
+//              (param $nwritten_out (@witx pointer (@witx usize)))
+//              (param $generation_out (@witx pointer u32))
+//              (param $kv_error_out (@witx pointer $kv_error))
+//              (result $err (expected (error $fastly_status)))
+//          )
+
+//go:wasmimport fastly_kv_store lookup_wait
+//go:noescape
+func fastlyKVStoreLookupWait(
+	h kvstoreLookupHandle,
 	b prim.Pointer[bodyHandle],
+	metaData prim.Pointer[prim.U8], metaLen prim.Usize,
+	nwritten prim.Pointer[prim.Usize],
+	generation prim.Pointer[prim.U32],
+	kvErr prim.Pointer[KVError],
 ) FastlyStatus
 
-// Lookup returns the value for key, if it exists.
-func (o *KVStore) Lookup(key string) (io.Reader, error) {
+// LookupWait returns a lookup response for a pending lookup handle
+func (kv *KVStore) LookupWait(lookuph kvstoreLookupHandle) (KVLookupResult, error) {
 	body := HTTPBody{h: invalidBodyHandle}
 
-	keyBuffer := prim.NewReadBufferFromString(key).Wstring()
+	meta := prim.NewWriteBuffer(kvstoreMetadataMaxBufLen)
+	var generation prim.U32
 
-	if err := fastlyObjectStoreLookup(
-		o.h,
-		keyBuffer.Data, keyBuffer.Len,
+	var kvErr KVError = KVErrorUninitialized
+
+	if err := fastlyKVStoreLookupWait(
+		lookuph,
 		prim.ToPointer(&body.h),
+		prim.ToPointer(meta.U8Pointer()),
+		meta.Cap(),
```

Comment:
```suggestion
		prim.ToPointer(meta.U8Pointer()), meta.Cap(),
```
To keep line groupings as in the `func` declaration.